File: nflow-engine/src/main/java/com/nitorcreations/nflow/engine/workflow/StateExecution.java
package com.nitorcreations.nflow.engine.workflow;

import org.joda.time.DateTime;

public interface StateExecution {

  String getBusinessKey();

  String getRequestData();
  <T> T getRequestData(Class<T> type);

  int getRetries();
  boolean isFailure();

  String getVariable(String name);
  <T> T getVariable(String name, Class<T> type);
  String getVariable(String name, String defaultValue);
  void setVariable(String name, String value);
  void setVariable(String name, Object value);

  void setNextActivation(DateTime time);
  void setNextState(WorkflowState state);
  void setNextStateReason(String stateText);
  void setNextState(WorkflowState state, String stateText, DateTime time);
  void setFailure(boolean failure);
  void setSaveTrace(boolean saveTrace);

}

File: nflow-examples/spring-boot/full-stack-kotlin/src/engine/StateExecution.kt
package nflow.kotlin.engine

import io.nflow.engine.workflow.definition.StateExecution

inline fun <reified TYPE : Any> StateExecution.getVar(name: String): TYPE =
        getVariable(name, TYPE::class.java)

inline fun <reified TYPE : Any> StateExecution.getVar(name: String, defaultValue: TYPE): TYPE =
        getVariable(name, TYPE::class.java, defaultValue)